### PR TITLE
Small RFlags fixes

### PR DESCRIPTION
**Pull request by ret2libc**
Fix some memory leaks by using ht_update_key API and makes code more consisent. All changes to name/offset of a flag pass through some helper functions that adjust the related storage. When you change name you need to adjust the Ht, when you change offset you need to adjust the by_off skiplist.

### CODE REVIEW

File: libr/flag/flag.c [hunk 1]
```diff
@@ -219,11 +298,11 @@ static bool print_flag_json(RFlagItem *flag, void *user) {
 		return true;
 	}
 	u->f->cb_printf ("%s{\"name\":\"%s\",\"size\":%" PFMT64d ",",
-		      u->first? "": ",", flag->name, flag->size);
+		u->first? "": ",", flag->name, flag->size);
 	if (flag->alias) {
 		u->f->cb_printf ("\"alias\":\"%s\"", flag->alias);
 	} else {
-		u->f->cb_printf ("\"offset\":%"PFMT64d, flag->offset);
+		u->f->cb_printf ("\"offset\":%" PFMT64d, flag->offset);
```
Comment (Maijin) [inline libr/flag/flag.c]: pj.c?

Comment (ret2libc) [reply libr/flag/flag.c]: what?

**Comment by ret2libc** [comment]
r2r PR: https://github.com/radare/radare2-regressions/pull/1655

File: libr/flag/flag.c (hunk 1 above)
Comment (radare) [reply libr/flag/flag.c]: But this can be done in a separate pr

Comment (ret2libc) [reply libr/flag/flag.c]: done

**Comment by ret2libc** [comment]
@radare I think this PR is ready to be merged. Let's wait for travis again though.

**Comment by radare** [comment]
me  too

**Pull request merged**